experimental/devices/sn3218/sn3218.go

```diff
@@ -0,0 +1,136 @@
+// Copyright 2016 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package sn3218
+
+import (
+	"errors"
+
+	"periph.io/x/periph/conn/i2c"
+)
+
+const (
+	i2cAddress             = 0x54
+	cmdEnableOutput        = 0x00
+	cmdSetBrightnessValues = 0x01
+	cmdEnableLeds          = 0x13
+	cmdUpdate              = 0x16
+	cmdReset               = 0x17
+)
+
+type dev struct {
+	i2c        *i2c.Dev
+	states     [18]bool
+	brightness [18]byte
+}
+
+// New returns an object that handles communication with a SN3218 LED driver
+// The SN3213 only accepts writes, so this object also keeps the state of
+// each LED.
+// Important: The device is write only, so it is not possible to read the state
+// of each LED. Therefore it's recommended to run d.Reseet() after this, so that
+// all LEDs are off and have the brightness 0.  Otherwise GetLedState() returns
+// wrong results.
+func New(bus i2c.Bus) (*dev, error) {
+	d := &i2c.Dev{Bus: bus, Addr: i2cAddress}
```

Comment:
Since it's a write-only device, I'd vote to reset it. It's annoying in some cases, as some user may want to "keep the previous state as-is" but I'd prefer to err on the safer side here and reset automatically instead of having to comment everywhere that the user has to call Reset().

Reply:
done